lib/cloudant.js

```diff
@@ -524,20 +530,37 @@ Cloudant.prototype.updateAll = function(model, where, data, options, cb) {
           model, where, data, options);
   var self = this;
   var mo = self.selectModel(model);
+  var resultArray = [];
   self.all(model, {where: where}, {raw: true}, function(err, docs) {
     if (err) return cb(err, docs);
-    if (docs.length === 0) return cb(null, {count: 0});
+    if (docs.length === 0) {
+      err = {};
+      err.statusCode = 404;
+      err.message = g.f('No document found with filter %j', where);
+      return cb(err);
+    }
 
-    data = getPlainJSONData.call(self, model, data);
     async.each(docs, function(doc, cb) {
+      if (data._rev) {
+        if (doc._rev !== data._rev) {
+          err = {};
+          err.statusCode = 409;
+          err.message = g.f('Document update conflict with rev %s', doc._rev);
```

Comment:
Returning an error via `async.each` would mean that the user/app making these calls might have to repeat this a ridiculous number of times to discover all of the individual document conflicts, since the first failure will halt the remainder of these calls.

I would expect that before you attempt the asynchronous `updateAttributes` call, that you would compare the `_.rev` in a loop.
```
var conflicts = [];
_.each(docs, function(doc) {
  var item = _.find(data, { id: doc.id });
  if (item._rev !== data._rev) {
    conflicts.push(doc.id);
  }
});

if (!_.isEmpty(conflicts)) {
  err = new Error(util.format('Document conflict(s):%s%s', os.EOL, conflicts.join(os.EOL));
  err.statusCode = 409;
  return cb(err);
}
// Continue on with the updateAttributes calls...
```